pandas/tests/frame/test_convert_to.py

```diff
@@ -190,17 +147,65 @@ def test_to_records_with_unicode_column_names(self):
         )
         tm.assert_almost_equal(result, expected)
 
+    @pytest.mark.parametrize('mapping', [
+        dict,
+        collections.defaultdict(list),
+        collections.OrderedDict])
+    def test_to_dict(self, mapping):
```

Comment:
This may be tested elsewhere, but can you add a test with a dataframe that has duplicate columns? Make sure to catch the warning.